src/main/java/sparksoniq/jsoniq/item/StringItem.java

```diff
@@ -76,6 +80,25 @@ public boolean isTypeOf(ItemType type) {
         return type.getType().equals(ItemTypes.StringItem) || super.isTypeOf(type);
     }
 
+    @Override
+    public boolean isCastableAs(AtomicType type) {
+        if (type.getType() == AtomicTypes.StringItem)
+            return true;
+        try {
+            if (type.getType() == AtomicTypes.IntegerItem)
+                Integer.parseInt(this._value);
```

Comment:
Are you positive the JVM optimizer will not throw this line away, because the result is not assigned? If we can't tell whether Java guarantees that, I can still live with that because there are tests in place, but we need to keep in mind that may change at any time.

Reply:
I think the JVM could skip them if those parsing method were not called outside a try-catch. Since we are catching an exception (_NumberFormatException_) that they all might potentially throw, the JVM should not skip those statement. This is also what the IDE seems to suggest me if I were to take those method calls out of the try-catch, by saying that their result is not used.